### PR TITLE
Kiali Istio dashboards incompatible with thanos-query

### PR DESCRIPTION
**Pull request by aljesusg**
Fix https://github.com/kiali/kiali/issues/4303

PR operator : https://github.com/kiali/kiali-operator/pull/427

### CODE REVIEW

**Review by jmazzitelli — CHANGES_REQUESTED** [review]
comment from the issue:
> I'm going to create the PR with these proposed changes but I think as far as I know that this is not going to fix it.

If we don't think this is going to do anything, then we should not merge this. If it doesn't fix the problem, we should not add this "just to add it". We should look into things further and fix the issue and not add code that doesn't address the problem.

**Comment by jmazzitelli** [comment]
I would set this PR to "draft" status, build the server with these changes, and ask for the person who reported the issue to test with that build and see if it does, indeed, address the issue.

**Comment by aljesusg** [comment]
Totally agree, I am trying to test this in GCp but IAM is playing with me and I can’t write in bucket, but as you said maybe @vbergbauer can test it

**Comment by vbergbauer** [comment]
@aljesusg : If you build a container image for me, I'll happily test it.
I know how to reproduce the problem on various clusters (KinD, GKE, OCP).

**Comment by aljesusg** [comment]
Thanks @vbergbauer I pushed the image in quay.io/aljesusg/kiali:dev ...I am checking and I think it solves it

File: config/config.go [hunk 1]
```diff
@@ -106,6 +106,13 @@ func (a *Auth) Obfuscate() {
 	a.CAFile = "xxx"
 }
 
+// Thanos describes configuration of the Thanos proxy component
+type Thanos struct {
```
Comment (jmazzitelli) [inline config/config.go:110]: shouldn't we call this ThanosProxy just to remain consistent with the json name?

**Comment by jmazzitelli** [comment]
I'm not sure if anything else has to change, but I will bring this up ... there is another place where prometheus config is in our configuration (this is Kiali CR's `spec.external_services.custom_dashboards`):

```
// CustomDashboardsConfig describes configuration specific to Custom Dashboards
type CustomDashboardsConfig struct {
	DiscoveryEnabled       string           `yaml:"discovery_enabled,omitempty"`
	DiscoveryAutoThreshold int              `yaml:"discovery_auto_threshold,omitempty"`
	Enabled                bool             `yaml:"enabled,omitempty"`
	IsCore                 bool             `yaml:"is_core,omitempty"`
	NamespaceLabel         string           `yaml:"namespace_label,omitempty"`
	Prometheus             PrometheusConfig `yaml:"prometheus,omitempty"`
}
```

Notice there is a PrometheusConfig here also (it defines the secondary Prometheus that Kiali can query for the custom dashboard metrics.) This means there can also be a ThanosProxy defined here for the secondary Prometheus. It is probably a no-op (i.e. if a user specifies a ThanosProxy here inside this `spec.external_services.custom_dashboards.prometheus.thanos_proxy` it will have no effect and be ignored).

I want to mention this in case we actually do need to do something here with ThanosProxy config when accessing Prometheus for custom dashboard metrics.

**Comment by aljesusg** [comment]
Yes If this work correctly I'll do the changes

File: config/config.go [hunk 2]
```diff
@@ -468,6 +476,11 @@ func NewConfig() (c *Config) {
 				Enabled:                true,
 				IsCore:                 false,
 				NamespaceLabel:         "kubernetes_namespace",
+				Prometheus: PrometheusConfig{
+					ThanosProxy: ThanosProxy{
+						Enabled: false
+					}
+				}
```
Comment (jmazzitelli) [inline config/config.go:483]: This should fix the linter error that is causing the build to fail:
```suggestion
						Enabled: false,
					},
				},
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Comment by jmazzitelli** [comment]
If this is good and will be merged, we'll need an operator PR.

* Add to example CR: https://github.com/kiali/kiali-operator/blob/v1.40.0/deploy/kiali/kiali_cr.yaml#L619
* Add defaults: https://github.com/kiali/kiali-operator/blob/v1.40.0/roles/default/kiali-deploy/defaults/main.yml#L178

**Comment by vbergbauer** [comment]
Hi @aljesusg 
Sorry for the delay (busy week last week).
I tested quay.io/aljesusg/kiali:dev and it does resolve the issue with custom dashboards without resorting to my workaround.
Thank you.

**Comment by aljesusg** [comment]
Great we can continue @jmazzitelli

**Comment by jshaughn** [comment]
@aljesusg Same comment as in the sibling PR.

**Pull request merged**